code/dynamic_programming/factorial/factorial.cpp

```diff
@@ -0,0 +1,31 @@
+/* 
+ * C++ Program to Find Factorial of a Number using Dynamic Programming 
+ */
+#include <iostream>
+#define ll long long
+using namespace std;
+ 
+ll result[50] = {0};
+/* 
+ * Find Factorial of a Number using Dynamic Programming 
+ */
+ll Factorial(int n)
```

Comment:
The return type is somewhat arbitrary. Since the type argument is `int`, why not return an `int` or vice versa for long?